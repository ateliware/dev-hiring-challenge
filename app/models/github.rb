class Github
  include HTTParty
  base_uri 'api.github.com'

  def initialize
    @auth = { headers: { "Authorization" => "token #{ENV['GITHUB_TOKEN']}" } }
  end

  def user
    res = self.class.get('/user', @auth)
  end

  def search_repositories
    res = self.class.get('/search/repositories?q=language:ruby+language:javascript+language:swift+language:elixir+language:c', @auth)
  end

  def repositories(id)
    res = self.class.get("/repositories/#{id}", @auth)
  end
end
