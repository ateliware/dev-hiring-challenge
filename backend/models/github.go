package models

import "time"

type Repositorio struct {
	Nome                  string    `json:"nome"                    example:"LeetCodeAnimation"`
	Linguagem             string    `json:"linguagem"               example:"Java"`
	Descricao             string    `json:"descricao"               example:"Demonstrate all the questions on LeetCode in the form of animation."`
	DataCriacao           time.Time `json:"data_criacao"            example:"2018-12-06T08:01:22Z"`
	DataUltimaAtualizacao time.Time `json:"data_ultima_atualizacao" example:"2022-06-07T19:52:01Z"`
	Tamanho               int       `json:"tamanho"                 example:"509807"`
	Estrelas              int       `json:"estrelas"                example:"70436"`
	Forks                 int       `json:"forks"                   example:"13535"`
	IssuesAbertas         int       `json:"issues_abertas"          example:"13"`
	OwnerNome             string    `json:"owner_nome"              example:"MisterBooo"`
	OwnerTipo             string    `json:"owner_tipo"              example:"User"`
	OwnerURL              string    `json:"owner_url"               example:"https://github.com/MisterBooo"`
} // @name Repositorio
