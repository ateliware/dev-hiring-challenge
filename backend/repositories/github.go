package repositories

import (
	"desafio_ateliware/backend/database"
	"desafio_ateliware/backend/models"
)

type RepositoryGitHub struct{}

func (RepositoryGitHub) SalvarRepositorios(repositorios []models.Repositorio) error {
	db := database.SetupDB()
	defer db.Close()

	db = db.Begin()

	for i := range repositorios {

		err := db.
			Table("repositorio").
			Create(&repositorios[i]).
			Error
		if err != nil {
			db.Rollback()
			return err
		}

	}

	if err := db.Commit().Error; err != nil {
		return err
	}

	return nil
}

func (RepositoryGitHub) ExcluirRepositorios(linguagem string) error {
	db := database.SetupDB()
	defer db.Close()

	err := db.
		Table("repositorio").
		Where("lower(linguagem) = lower(?)", linguagem).
		Delete(nil).
		Error

	return err
}

func (RepositoryGitHub) BuscarRepositorios(linguagem string) ([]models.Repositorio, error) {
	db := database.SetupDB()
	defer db.Close()

	var repositorios []models.Repositorio
	err := db.
		Table("repositorio").
		Where("lower(linguagem) = lower(?)", linguagem).
		Order("estrelas DESC").
		Find(&repositorios).
		Error

	return repositorios, err
}
