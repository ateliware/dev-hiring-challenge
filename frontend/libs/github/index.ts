import React from "react"
import { Paginator } from "../paginator";
import { ApiResponse, FeedbackMessage, getApiFeedback, isApiLoaded, isApiLoading, isApiWaiting, RemoteDataState, Repositorie, setAsError, setLoaded, setLoading } from "../shared"
import { get } from "../shared/api";

export enum Language {
  PYTHON=0,
  JAVASCRIPT,
  RUST,
  C,
  HASKELL
}

export interface GithubRepositoriesParams {
  label: string;
  language: Language;
}

export interface GithubRepositoriesResponse {
  incomplete_results: boolean;
  items: Repositorie[];
  total_count: number;
}

interface GithubRepositoriesHook {
  repositories: ApiResponse<Repositorie[]>;
  searchParams: GithubRepositoriesParams;
  searchRepositories: (paginator: Paginator, updatePaginator: (nextPaginator: Partial<Paginator>) => void) => Promise<void>;
  setSearchParams: React.Dispatch<React.SetStateAction<GithubRepositoriesParams>>;
}

export const useGithubRepositories = (initialSearchParams: GithubRepositoriesParams): GithubRepositoriesHook => {
  const [ repositories, setRepositories ] = React.useState<ApiResponse<Repositorie[]>>(initialRepositories);
  const [ searchParams, setSearchParams ] = React.useState<GithubRepositoriesParams>(initialSearchParams);

  const setReposLoading = setLoading(setRepositories);
  const setReposLoaded = setLoaded(setRepositories);
  const setReposAsError = setAsError(setRepositories);

  const searchRepositories = async (
    paginator: Paginator,
    updatePaginator: (nextPaginator: Partial<Paginator>) => void
  ): Promise<void> => {
    setReposLoading();
    const reposResponse = await requestGithubRepositories({
      ...searchParams,
      language: languageToLabel(searchParams.language).toLowerCase(),
      page: paginator.page.toString()
    });
    switch(reposResponse.state) {
      case(RemoteDataState.LOADED):
        setReposLoaded(reposResponse.data.items, "Repositories loaded.");
        updatePaginator({
          totalCount: reposResponse.data.total_count
        });
        break;
      case(RemoteDataState.ERROR):
        setReposAsError(reposResponse.detail);
        break;
    }
  }

  return {
    repositories,
    searchParams,
    searchRepositories,
    setSearchParams,
  }
}

export const selectors = {
  isLoading: (state?: GithubRepositoriesHook): boolean => isApiLoading(state?.repositories),
  isWaiting: (state?: GithubRepositoriesHook): boolean => isApiWaiting(state?.repositories),
  isLoaded: (state?: GithubRepositoriesHook): boolean => isApiLoaded(state?.repositories),
  getFeedback: (state?: GithubRepositoriesHook): FeedbackMessage | null => getApiFeedback(state?.repositories),
};

export const stringToLanguage = (value: string): Language => {
  const lang = ({
    '0': Language.PYTHON,
    '1': Language.JAVASCRIPT,
    '2': Language.RUST,
    '3': Language.C,
    '4': Language.HASKELL,
  })[value];

  return lang || Language.PYTHON;
}

export const languageToLabel = (lang: Language): string => {
  return ({
    [Language.PYTHON]: 'Python',
    [Language.JAVASCRIPT]: 'Javascript',
    [Language.RUST]: 'Rust',
    [Language.C]: 'C',
    [Language.HASKELL]: 'Haskell',
  })[lang];
}

const requestGithubRepositories = get<GithubRepositoriesResponse>("/api/v1/github/repositories/");

const initialRepositories: ApiResponse<Repositorie[]> = {
  state: RemoteDataState.WAITING,
}
