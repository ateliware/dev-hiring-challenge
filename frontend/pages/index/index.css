* {
    font-family: -apple-system, BlinkMacSystemFont, “Segoe UI”, Roboto, Oxygen-Sans, Ubuntu, Cantarell, Helvetica, Arial, sans-serif !important;
    text-align: center;
}

a {
    text-decoration: none !important;
}

a:hover {
    text-decoration: underline !important;
}

select {
    text-align: start !important;
    height: calc(3.5rem + 10px) !important;
}

option,
.toast-body {
    text-align: start !important;
}

.toast {
    border: 0px !important;
}

.toast-title-icon::before {
    display: inline-block;
    min-height: 12px;
    min-width: 12px;
    margin-right: 4px;
    margin-bottom: -1px;
    border-radius: 8px;
    content: "";
}

.toast-title-icon.success::before {
    background: green;
}

.toast-title-icon.error::before {
    background: red;
}

th.col-actions {
    width: 80px !important;
}

td.col-actions {
    width: 80px !important;
    display: flex;
    justify-content: center;
    align-items: center;
}

.col-actions button {
    border: none;
    padding: 3px 6px;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 4px;
}

.modal-body .row div {
    display: flex;
    justify-content: flex-start;

}

.modal-body .row div b {
    margin-right: 4px;
}

/***** Loader *****/
body.loader {
    overflow: hidden !important;
}

div.loader {
    position: fixed;
    top: 0;
    left: 0;
    background: rgb(0 0 0 / 50%);
    z-index: 99996;
    width: 100vw;
    height: 100vh;
}

div.spinner-wrapper {
    border-radius: 8px;
    z-index: 99997;
    display: block;
    position: fixed;
    background-color: #FFFFFF;
    height: 130px;
    width: 130px;
    top: calc(50% - 65px);
    left: calc(50% - 65px);
}

p.spinner-label {
    -webkit-user-select: none;
    user-select: none;
    z-index: 99998;
    margin-top: 97px;
    text-align: center;
}

div.spinner {
    display: block;
    position: fixed;
    z-index: 99998;
    top: calc(50% - 42px);
    left: calc(50% - 30px);
    height: 60px;
    width: 60px;
    border: solid 4px transparent;
    border-top-color: #000000;
    border-left-color: #000000;
    border-radius: 100px;
    -webkit-animation: loader-spinner 1000ms linear infinite;
    -moz-animation: loader-spinner 1000ms linear infinite;
    -ms-animation: loader-spinner 1000ms linear infinite;
    -o-animation: loader-spinner 1000ms linear infinite;
    animation: loader-spinner 1000ms linear infinite;
}

@-webkit-keyframes loader-spinner {
    0% {
        -webkit-transform: rotate(0deg);
        transform: rotate(0deg);
    }
    100% {
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}

@-moz-keyframes loader-spinner {
    0% {
        -moz-transform: rotate(0deg);
        transform: rotate(0deg);
    }
    100% {
        -moz-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}

@-o-keyframes loader-spinner {
    0% {
        -o-transform: rotate(0deg);
        transform: rotate(0deg);
    }
    100% {
        -o-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}

@-ms-keyframes loader-spinner {
    0% {
        -ms-transform: rotate(0deg);
        transform: rotate(0deg);
    }
    100% {
        -ms-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}

@keyframes loader-spinner {
    0% {
        transform: rotate(0deg);
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
        transform: rotate(360deg);
    }
}

/***** Mobile fixes *****/

@media (max-width: 576px) {
    .col-sequencial-number,
    .col-actions {
        display: none;
    }
    .fade {
        opacity: 1px !important;
    }
    .table>:not(caption)>*>* {
        padding: 0.5rem 0.3rem !important;
        font-size: 14px !important;
    }
    select {
        text-align: center !important;
        height: calc(3.5rem + 25px) !important;
    }
    option {
        text-align: center !important;
    }
    .form-floating>.form-select {
        padding-top: 2.925rem !important
    }
}