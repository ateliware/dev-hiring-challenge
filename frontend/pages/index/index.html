<!DOCTYPE html>
<html lang="pt-BR">

<head>

    <!-- General -->
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta name="description" content="Veja os melhores repositórios das melhores linguagens de programação!">
    <meta name="author" content="Eduardo Mior">
    <meta name="robots" content="noindex, nofollow">
    <title>Top GitHub Repositórios 😎</title>

    <!-- Icons -->
    <link rel="icon" type="image/png" href="icon.png" sizes="192x192">
    <link rel="apple-touch-icon" type="image/png" href="apple-touch-icon.png" sizes="180x180">

    <!-- Android -->
    <meta name="theme-color" content="#FFFFFF">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="application-name" content="Top GitHub Repositórios 😎">

    <!-- iOS -->
    <meta name="apple-mobile-web-app-title" content="Top GitHub Repositórios 😎">
    <meta name="apple-mobile-web-app-capable" content="yes">

    <!-- OG -->
    <meta property="og:locale" content="pt_BR">
    <meta property="og:type" content="website">
    <meta property="og:title" content="Top GitHub Repositórios 😎">
    <meta property="og:site_name" content="Top GitHub Repositórios 😎">
    <meta property="og:description" content="Veja os melhores repositórios das melhores linguagens de programação!">
    <meta property="og:url" content="https://url-aqui.com">
    <meta property="og:image" content="https://url-aqui.com/images/logo2.png">
    <meta property="og:image:type" content="image/png">
    <meta property="og:image:alt" content="Top GitHub Repositórios 😎">

    <!-- CSS -->
    <link rel="stylesheet" href="index.css" />
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css">

    <!-- JS -->
    <script src="https://browser.sentry-cdn.com/7.1.1/bundle.min.js"></script>
    <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/js/bootstrap.bundle.min.js"></script>
    <script src="index.js"></script>

</head>

<body>
    <div class="container">

        <div class="row d-flex justify-content-center">

            <div class="col-12 mt-3">
                <h1 class="mb-2">Top GitHub Repositórios 😎</h1>
            </div>

            <div class="col-12">
                <span>Veja em tempo real os melhores repositórios das melhores linguagens de programação da atualidade!</span>
            </div>

            <div class="col-12 d-flex justify-content-center mb-1 mt-3 mt-sm-5">
                <div class="form-floating col-8">
                    <select class="form-select form-select-lg" id="linguagem-programacao">
                        <option selected disabled></option>
                        <option value="java">☕ Java</option>
                        <option value="ocaml">🐫 OCaml</option>
                        <option value="lua">🌙 Lua</option>
                        <option value="python">🐍 Python</option>
                        <option value="ruby">💎 Ruby</option>
                    </select>
                    <label for="linguagem-programacao">Selecione a sua linguagem de programação favorita</label>
                </div>
            </div>

            <div class="col-12 mt-1 mt-sm-4" id="dados-linguagem" style="display: none;">
                <div class="row d-flex justify-content-between">
                    <div class="col-12 col-md-4 mt-3">
                        <button id="buscar-repositorios-github" type="button" class="col-12 btn btn-success"
                            title="Clique para buscar no GitHub a lista atualizada dos TOP repositórios da sua linguagem de programação favorita">🐙 Buscar dados no GitHub</button>
                    </div>
                    <div class="col-12 col-md-4 mt-3">
                        <button id="salvar-repositorios-em-cache" type="button" class="col-12 btn btn-primary" disabled
                            title="Clique para salvar no banco de dados a lista de repositórios buscados no GitHub">💾 Salvar dados no DB</button>
                    </div>
                    <div class="col-12 col-md-4 mt-3">
                        <button id="excluir-repositorios-do-cache" type="button" class="col-12 btn btn-danger" disabled
                            title="Clique para apagar do banco de dados toda a lista de repositórios buscados no GitHub">🗑 Limpar dados do DB</button>
                    </div>
                </div>
                <table class="table mt-1">
                    <thead>
                        <tr>
                            <th scope="col" class="col-sequencial-number">#</th>
                            <th scope="col" class="col-repository">Repositório</th>
                            <th scope="col" class="col-owner">Owner</th>
                            <th scope="col" class="col-creation-date">Data de Criação</th>
                            <th scope="col" class="col-stars">Stars</th>
                            <th scope="col" class="col-actions">Detalhes</th>
                        </tr>
                    </thead>
                    <tbody class="table-group-divider">
                    </tbody>
                </table>
            </div>

        </div>

    </div>
</body>

</html>
