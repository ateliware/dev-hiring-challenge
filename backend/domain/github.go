package domain

import "desafio_ateliware/backend/models"

type IRepoitoryGitHub interface {
	BuscarRepositorios(linguagem string) ([]models.Repositorio, error)

	SalvarRepositorios([]models.Repositorio) error

	ExcluirRepositorios(linguagem string) error
}
