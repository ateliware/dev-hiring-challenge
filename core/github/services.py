from requests import get

from core.github.models import Linguagem as LinguagemModel

from core.settings import URL_API_GITHUB


def fetch_repositorios_por_linguagem(linguagem: str):

    q = f"language:{linguagem.lower()}"

    parametros = {
        "q": q
    }

    url = f"{URL_API_GITHUB}/search/repositories"

    response = get(url, params=parametros)

    if response.status_code != 200:
        response.raise_for_status()
    
    json = response.json()

    return json["items"]


def fetch_repositorios():
    linguagens = LinguagemModel.objects.all()

    for linguagem in linguagens:
        repositorios = fetch_repositorios_por_linguagem(linguagem.nome)

        for repositorio in repositorios:
            yield {
                "nome": repositorio["name"],
                "url": repositorio["html_url"],
                "estrelas": repositorio["stargazers_count"],
                "acompanhadores": repositorio["watchers"],
                "pontos": repositorio["score"],
                "descricao": repositorio["description"],
                "linguagem": linguagem,
            }
