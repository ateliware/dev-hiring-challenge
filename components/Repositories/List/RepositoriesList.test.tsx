import { render, screen } from "@testing-library/react";
import { RepositoryJsonData } from "../../../services/github";
import { RepositoriesList } from "./index";

const MockRepositoriesData = () => {
  return [
    {
      created_at: "2018-10-07T19:32:59Z",
      description:
        "A predictable state management library that helps implement the BLoC design pattern",
      full_name: "felangel/bloc",
      html_url: "https://github.com/felangel/bloc",
      id: 151977818,
      language: "Dart",
      name: "bloc",
      node_id: "MDEwOlJlcG9zaXRvcnkxNTE5Nzc4MTg=",
      owner: {
        login: "felangel",
        id: 8855632,
        url: "https://api.github.com/users/felangel",
        avatar_url: "https://avatars.githubusercontent.com/u/8855632?v=4",
      },
      stargazers_count: 8538,
      topics: [
        "angulardart",
        "bloc",
        "concurrency",
        "dart",
        "dart-library",
        "dart-web",
        "dartlang",
        "flutter",
        "flutter-package",
        "library",
        "state-management",
      ],
      updated_at: "2022-02-14T16:37:28Z",
      url: "https://api.github.com/repos/felangel/bloc",
    },
    {
      created_at: "2018-04-20T04:33:43Z",
      description:
        "A powerful Http client for Dart, which supports Interceptors, FormData, Request Cancellation, File Downloading, Timeout etc.",
      full_name: "flutterchina/dio",
      html_url: "https://github.com/flutterchina/dio",
      id: 130309267,
      language: "Dart",
      name: "dio",
      node_id: "MDEwOlJlcG9zaXRvcnkxMzAzMDkyNjc=",
      owner: {
        login: "flutterchina",
        id: 38549573,
        url: "https://api.github.com/users/flutterchina",
        avatar_url: "https://avatars.githubusercontent.com/u/38549573?v=4",
      },
      stargazers_count: 10397,
      topics: ["dart", "flutter"],
      updated_at: "2022-02-14T17:37:33Z",
      url: "https://api.github.com/repos/flutterchina/dio",
    },
  ] as RepositoryJsonData[];
};

describe("RepositoriesListItem", () => {
  const thisReposData = MockRepositoriesData();

  it("renderiza os elementos da lista", () => {
    const wrapper = render(<RepositoriesList data={thisReposData} />);

    const firstRepoHeading = screen.getByRole("heading", {
      name: thisReposData[0].name,
    });
    expect(firstRepoHeading).toBeInTheDocument();

    const secondRepoHeading = screen.getByRole("heading", {
      name: thisReposData[1].name,
    });
    expect(secondRepoHeading).toBeInTheDocument();

    expect(wrapper.container).toMatchSnapshot();
  });
});
