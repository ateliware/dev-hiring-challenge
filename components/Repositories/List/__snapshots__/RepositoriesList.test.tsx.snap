// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesListItem renderiza os elementos da lista 1`] = `
<div>
  <ul
    class="main"
  >
    <li
      class="main"
    >
      <header
        class="header"
      >
        <a
          href="https://github.com/felangel/bloc"
          rel="noreferrer"
          target="_blank"
        >
          <h4
            class="title"
          >
            bloc
          </h4>
        </a>
      </header>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Estrelas
        </span>
        <span
          class="sectionValue"
        >
          8538
        </span>
      </section>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Proprietario
        </span>
        <span
          class="sectionValue"
        >
          <div
            class="ownerInfo"
          >
            <img
              alt="felangel"
              src="https://avatars.githubusercontent.com/u/8855632?v=4"
            />
            <span>
              felangel
            </span>
          </div>
        </span>
      </section>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Descrição
        </span>
        <span
          class="sectionValue"
        >
          A predictable state management library that helps implement the BLoC design pattern
        </span>
      </section>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Tópicos
        </span>
        <span
          class="sectionValue"
        >
          angulardart, bloc, concurrency, dart, dart-library, dart-web, dartlang, flutter, flutter-package, library, state-management
        </span>
      </section>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Linguagem
        </span>
        <span
          class="sectionValue"
        >
          Dart
        </span>
      </section>
    </li>
    <li
      class="main"
    >
      <header
        class="header"
      >
        <a
          href="https://github.com/flutterchina/dio"
          rel="noreferrer"
          target="_blank"
        >
          <h4
            class="title"
          >
            dio
          </h4>
        </a>
      </header>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Estrelas
        </span>
        <span
          class="sectionValue"
        >
          10397
        </span>
      </section>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Proprietario
        </span>
        <span
          class="sectionValue"
        >
          <div
            class="ownerInfo"
          >
            <img
              alt="flutterchina"
              src="https://avatars.githubusercontent.com/u/38549573?v=4"
            />
            <span>
              flutterchina
            </span>
          </div>
        </span>
      </section>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Descrição
        </span>
        <span
          class="sectionValue"
        >
          A powerful Http client for Dart, which supports Interceptors, FormData, Request Cancellation, File Downloading, Timeout etc.
        </span>
      </section>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Tópicos
        </span>
        <span
          class="sectionValue"
        >
          dart, flutter
        </span>
      </section>
      <section
        class="section"
      >
        <span
          class="sectionLabel"
        >
          Linguagem
        </span>
        <span
          class="sectionValue"
        >
          Dart
        </span>
      </section>
    </li>
  </ul>
</div>
`;
